c2corg_api/views/waypoint.py

```diff
@@ -28,7 +29,11 @@
 from functools import lru_cache
 from sqlalchemy.orm import joinedload, load_only
 from sqlalchemy.orm.util import aliased
-from sqlalchemy.sql.expression import and_
+from sqlalchemy.sql.elements import literal_column
+from sqlalchemy.sql.expression import and_, union
+
+# the number of routes that are included for waypoints
+NUM_ROUTES = 30
```

Comment:
> And these pages take 30 seconds to load, they are very unpractical because you are not able to find a route in a list of 100+ entries and they would take a ridiculous amount of space with the card layout. For me it is not an option to show 100+ routes on one page.

OK thanks for the arguments.

> I don't think it is important which routes you show if you have 30+ routes. If you have more than 30 you click on the link and then you can better browse them.

OK. Then let's keep your PR as is.
I will create an issue in the UI since we need a "show more" button for routes the same way it is done for outings.